Tribler/community/multichain/community.py

```diff
@@ -293,34 +293,97 @@ def get_statistics(self, public_key=None):
+        latest block from the database is retrieved.
+
+        :param public_key: the public key for which a node dictionary has to be created
+        :param nodes: the dictionary of currently recorded nodes
+        :param total_up: the total up amount
+        :param total_down: the total down amount
+        :return: a dictionary corresponding to the node in the correct format
+        """
+        if public_key in nodes:
+            return {"public_key": public_key, "total_up": max(total_up, nodes[public_key]["total_up"]),
+                    "total_down": max(total_down, nodes[public_key]["total_down"])}
+        else:
+            if total_up and total_down:
+                return {"public_key": public_key, "total_up": total_up, "total_down": total_down}
+            else:
+                unhex_pk = unhexlify(public_key)
+                total_traffic = self.persistence.total_traffic(unhex_pk)
+                latest_block = self.persistence.get_latest(unhex_pk)
+                if latest_block:
+                    return {"public_key": public_key, "total_up": max(latest_block.total_up, total_traffic[0]),
+                            "total_down": max(latest_block.total_down, total_traffic[1])}
+                else:
+                    return {"public_key": public_key, "total_up": total_traffic[0], "total_down": total_traffic[1]}
+
+    @staticmethod
+    def update_edges(from_pk, to_pk, edges, amount=0):
+        """
+        Update the given edge dictionary with the newly acquired data.
+
+        If there is not yet an entry for the public key pair, it is automatically added to the edges dictionary. If
+        an edge is already recorded, the given amount is added to the total
+
+        :param from_pk: the public key of the node where the edge originates from
+        :param to_pk: the public key of the node where the edge arrives
+        :param edges: the dictionary of edges
+        :param amount: the amount of data transferred over this edge
+        """
+        if from_pk in edges:
+            edges[from_pk][to_pk] = edges[from_pk][to_pk] + amount if to_pk in edges[from_pk] else amount
+        else:
+            edges[from_pk] = {to_pk: amount}
+
     @blocking_call_on_reactor_thread
     def get_graph(self, public_key=None, neighbor_level=1):
         """
         Return a dictionary with the neighboring nodes and edges of a certain focus node within a certain radius,
-        regarding the local multichain database.
+        regarding the local MultiChain database.
 
         :param public_key: the public key of the focus node in raw format
         :param neighbor_level: the radius within which the neighbors have to be returned
-        :return: a tuple of a dictionary with nodes and a dictionary with edges
+        :return: a tuple of a list with nodes and a list with edges
         """
         if public_key is None:
             public_key = self.my_member.public_key
```

Comment:
This should not be done as self.my_member.public_key is subject to change. This change would then not be used in this function.